src/scripts/browserDeviceProfile.js

```diff
@@ -474,12 +484,15 @@ export default function (options) {
 
     if (supportsMp3VideoAudio) {
         videoAudioCodecs.push('mp3');
+    }
 
-        // PS4 fails to load HLS with mp3 audio
-        if (!browser.ps4) {
-            hlsInTsVideoAudioCodecs.push('mp3');
-        }
+    // Safari supports mp3 with HLS, but only in mpegts container, and the supportsMp3VideoAudio will return false.
+    if (browser.safari || supportsMp3VideoAudio) {
```

Comment:
> I suppose supportsMp3VideoAudio is true for Safari and Play Station

No, it is not.

Reply:
Didn't it add MP3 to TS for Safari before?

`supportsMp3VideoAudio` is "MP3 in MP4", which was used to add MP3 to TS and fMP4.

Does Safari play MP3 in MP4 (direct, non-HLS)?

Reply:
> Didn't it add MP3 to TS for Safari before?

No, it always transcodes. `supportsMp3VideoAudio` is false on Safari.

> supportsMp3VideoAudio is "MP3 in MP4", which was used to add MP3 to TS and fMP4.

No, it should not be. It should only be considered as direct play because the testing this function does has nothing to do with HLS, and is very wrong for HLS.

> Does Safari play MP3 in MP4 (direct, non-HLS)?

No, there will be no sound.

Reply:
This is exactly why I said this check is too naive because we indeed have too many false assumptions.

Reply:
> No, it always transcodes. `supportsMp3VideoAudio` is false on Safari.

👍

> No, it should not be. It should only be considered as direct play because the testing this function does has nothing to do with HLS, and is very wrong for HLS.

I am not saying it should, I am saying how it was. And that was a note, not a question.

> No, there will be no sound.

👍